File: src/main/java/com/turchinsky/entities/TicketEntity.java
package com.turchinsky.entities;

import javax.persistence.*;

@Entity
@Table(name = "ticket")
public class TicketEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id"/*, nullable = false, insertable = false, updatable = false*/)
    private int id;

    @Basic
    @Column(name = "price")
    private int price;

    @Basic
    @Column(name = "station_from")
    private int stationFrom;

    @Basic
    @Column(name = "station_to")
    private int stationTo;

    @Basic
    @Column(name = "line_id_from")
    private int lineIdFrom;

    @Basic
    @Column(name = "line_id_to")
    private int lineIdTo;

    @Basic
    @Column(name = "holder_id")
    private int holderId;

    @Basic
    @Column(name = "routegroupid")
    private int routeGroupId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "station_from", referencedColumnName = "id", insertable = false, updatable = false)
    private StationEntity refStationFrom;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "station_to", referencedColumnName = "id", insertable = false, updatable = false)
    private StationEntity refStationTo;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "line_id_from", referencedColumnName = "line_id", insertable = false, updatable = false)
    private ScheduleDetailsEntity refScheduleDetailsEntityFrom;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "line_id_to", referencedColumnName = "line_id", insertable = false, updatable = false)
    private ScheduleDetailsEntity refScheduleDetailsEntityTo;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "holder_id", referencedColumnName = "id", insertable = false, updatable = false)
    private TicketHolderEntity refHolderEntity;


    public TicketEntity() {
    }

    public TicketEntity(int price, int stationFrom, int stationTo, int lineIdFrom, int lineIdTo, int holderId,
                        int routeGroupId) {
        this.price = price;
        this.stationFrom = stationFrom;
        this.stationTo = stationTo;
        this.lineIdFrom = lineIdFrom;
        this.lineIdTo = lineIdTo;
        this.holderId = holderId;
        this.routeGroupId = routeGroupId;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    public int getStationFrom() {
        return stationFrom;
    }

    public void setStationFrom(int stationFrom) {
        this.stationFrom = stationFrom;
    }

    public int getStationTo() {
        return stationTo;
    }

    public void setStationTo(int stationTo) {
        this.stationTo = stationTo;
    }

    public int getLineIdFrom() {
        return lineIdFrom;
    }

    public void setLineIdFrom(int lineIdFrom) {
        this.lineIdFrom = lineIdFrom;
    }

    public int getLineIdTo() {
        return lineIdTo;
    }

    public void setLineIdTo(int lineIdTo) {
        this.lineIdTo = lineIdTo;
    }

    public ScheduleDetailsEntity getRefScheduleDetailsEntityFrom() {
        return refScheduleDetailsEntityFrom;
    }

    public void setRefScheduleDetailsEntityFrom(ScheduleDetailsEntity refScheduleDetailsEntityFrom) {
        this.refScheduleDetailsEntityFrom = refScheduleDetailsEntityFrom;
    }

    public ScheduleDetailsEntity getRefScheduleDetailsEntityTo() {
        return refScheduleDetailsEntityTo;
    }

    public void setRefScheduleDetailsEntityTo(ScheduleDetailsEntity refScheduleDetailsEntityTo) {
        this.refScheduleDetailsEntityTo = refScheduleDetailsEntityTo;
    }

    public int getHolderId() {
        return holderId;
    }

    public void setHolderId(int holderId) {
        this.holderId = holderId;
    }

    public StationEntity getRefStationFrom() {
        return refStationFrom;
    }

    public void setRefStationFrom(StationEntity refStationFrom) {
        this.refStationFrom = refStationFrom;
    }

    public StationEntity getRefStationTo() {
        return refStationTo;
    }

    public void setRefStationTo(StationEntity refStationTo) {
        this.refStationTo = refStationTo;
    }


    public int getRouteGroupId() {
        return routeGroupId;
    }

    public void setRouteGroupId(int groupId) {
        this.routeGroupId = groupId;
    }

    public TicketHolderEntity getRefHolderEntity() {
        return refHolderEntity;
    }

    public void setRefHolderEntity(TicketHolderEntity refHolderEntity) {
        this.refHolderEntity = refHolderEntity;
    }

}

File: src/main/java/com/turchinsky/kt/TicketEntity.kt
//package com.turchinsky.kt
//
//import javax.persistence.*
//
//@Entity
//@Table(name = "ticket", schema = "public", catalog = "sbb")
//open class TicketEntity {
//    @get:Id
//    @get:Column(name = "id", nullable = false)
//    var id: Int? = null
//
//    @get:Basic
//    @get:Column(name = "price", nullable = true)
//    var price: Int? = null
//
//    @get:Basic
//    @get:Column(name = "station_from", nullable = true, insertable = false, updatable = false)
//    var stationFrom: Int? = null
//
//    @get:Basic
//    @get:Column(name = "station_to", nullable = true, insertable = false, updatable = false)
//    var stationTo: Int? = null
//
//    @get:Basic
//    @get:Column(name = "line_id", nullable = true, insertable = false, updatable = false)
//    var lineId: Int? = null
//
//    @get:Basic
//    @get:Column(name = "passenger_id", nullable = true, insertable = false, updatable = false)
//    var passengerId: Int? = null
//
//    @get:ManyToOne(fetch = FetchType.LAZY)
//    @get:JoinColumn(name = "station_from", referencedColumnName = "id")
//    var refStationEntity: StationEntity? = null
//
//    @get:ManyToOne(fetch = FetchType.LAZY)
//    @get:JoinColumn(name = "station_to", referencedColumnName = "id")
//    var refStationEntity: StationEntity? = null
//
//    @get:ManyToOne(fetch = FetchType.LAZY)
//    @get:JoinColumn(name = "line_id", referencedColumnName = "line_id")
//    var refScheduleDetailsEntity: ScheduleDetailsEntity? = null
//
//    @get:ManyToOne(fetch = FetchType.LAZY)
//    @get:JoinColumn(name = "passenger_id", referencedColumnName = "id")
//    var refPassengerEntity: PassengerEntity? = null
//
//    override fun toString(): String =
//            "Entity of type: ${javaClass.name} ( " +
//                    "id = $id " +
//                    "price = $price " +
//                    "stationFrom = $stationFrom " +
//                    "stationTo = $stationTo " +
//                    "lineId = $lineId " +
//                    "passengerId = $passengerId " +
//                    ")"
//
//    // constant value returned to avoid entity inequality to itself before and after it's update/merge
//    override fun hashCode(): Int = 42
//
//    override fun equals(other: Any?): Boolean {
//        if (this === other) return true
//        if (javaClass != other?.javaClass) return false
//        other as TicketEntity
//
//        if (id != other.id) return false
//        if (price != other.price) return false
//        if (stationFrom != other.stationFrom) return false
//        if (stationTo != other.stationTo) return false
//        if (lineId != other.lineId) return false
//        if (passengerId != other.passengerId) return false
//
//        return true
//    }
//
//}
//
